src/BasisMatrices.jl

```diff
@@ -70,7 +70,8 @@ export BasisFamily, Cheb, Lin, Spline, Basis,
 # functions
 export nodes, get_coefs, funfitxy, funfitf, funeval, evalbase,
        derivative_op, row_kron, evaluate, fit!, update_coefs!,
-       complete_polynomial, complete_polynomial!, n_complete
+       complete_polynomial, complete_polynomial!, n_complete,
+       complete_polynomial_der, complete_polynomial_der!
```

Comment:
I was thinking that given that you've introduced a `Derivative{D}` type, that we could get away with only having the `complete_polynomial` and `complete_polynomial!` functions be exported. We could then infer if the user wanted a derivative based on whether or not a `Derivative` type was passed and/or a derivative keyword argument?